frame/babe/src/lib.rs

```diff
@@ -305,12 +305,31 @@ impl<T: Trait> Module<T> {
 		// epoch 0 as having started at the slot of block 1. We want to use
 		// the same randomness and validator set as signalled in the genesis,
 		// so we don't rotate the epoch.
-		now != sp_runtime::traits::One::one() && {
+		now != One::one() && {
 			let diff = CurrentSlot::get().saturating_sub(Self::current_epoch_start());
 			diff >= T::EpochDuration::get()
 		}
 	}
 
+	/// Return the _best guess_ block number, at which the next epoch change is predicted to happen.
+	///
+	/// In other word, this is only accurate if no slots are missed. Given missed slots, the slot
+	/// number will grow while the block number will not. Hence, the result can be interpreted as an
+	/// upper bound.
+	///
+	/// -------------- IMPORTANT NOTE --------------
+	/// This implementation is linked to how [`should_epoch_change`] is working. This might need to
+	/// be updated accordingly, if the underlying mechanics of slot and epochs change.
+	pub fn next_expected_epoch_change(now: T::BlockNumber) -> T::BlockNumber {
+		let next_slot = Self::current_epoch_start().saturating_add(T::EpochDuration::get());
+		let slots_remaining = next_slot
+			.checked_sub(CurrentSlot::get())
+			.unwrap_or(Bounded::max_value());
```

Comment:
It makes more sense given the caller. If we return zero, if the caller is comparing this value to be `le` of some threshold, then it will always think that the condition is met. While by returning max_value we say to the caller _infinite blocks are remaining_, so that no condition is triggered. 

But this is tailored for the staking use case. I will change to to returning just some option.

Reply:
I had this exact same question. Option does seem better.

Reply:
yes I marked this as resolved by mistake. Will refactor, option is better.

Reply:
I thought a bit more about it. 

if `checked_sub` fails here, it means a very severe bug in babe is happening. I will add a warning log to it, but returning None seems like a bit ignorant as well.

Reply:
Nonetheless, finally done.